glslc/src/file_compiler.cc

```diff
@@ -177,22 +170,28 @@ void FileCompiler::OutputMessages() {
   shaderc_util::OutputMessages(&std::cerr, total_warnings_, total_errors_);
 }
 
+namespace {
+// A helper function to return the base file name from either absolute path or
+// relative path representation of a file.
+// e.g.: dir_a/dir_b/file_c.vert => file_c.vert
+std::string GetBaseFileName(const std::string& file_path) {
+  return file_path.substr(file_path.find_last_of("/\\") + 1);
```

Comment:
**Actually** `npos` will be converted to `-1` only if `string::size_type` bit width equals `int` bit width, which isn't guaranteed by the standard.